src/tokens.js

```diff
@@ -1132,6 +1132,103 @@ const tokens = {
     return uid === HATHOR_TOKEN_CONFIG.uid;
   },
 
+  /**
+   * Checks if this transaction is the creation of an NFT following the NFT Standard Creation.
+   * @see https://github.com/HathorNetwork/rfcs/blob/master/text/0032-nft-standard.md#transaction-standard
+   * @param {Object} tx Transaction with decoded inputs and outputs
+   */
+  isNFTToken(tx) {
+    // Basic input validation
+    if (!tx) {
+      return false;
+    }
+
+    // If this is not a token creation tx, it cannot be an NFT
+    if (tx.version !== CREATE_TOKEN_TX_VERSION) {
+      return false;
+    }
+
+    // NFT creation must have at least a DataScript output (the first one) and a Token P2PKH output
+    if (tx.outputs.length < 2) {
+      return false;
+    }
+
+    // NFT creation DataScript output must have value 1 and must be of HTR
+    const firstOutput = tx.outputs[0];
+    if (firstOutput?.value !== 1 || firstOutput?.token_data !== 0) {
+      return false;
+    }
+    // First output must also have a standard-compliant script
+    if (!hasStandardScript(firstOutput)) {
+      return false;
+    }
+
+    // Validating the remaining outputs
+    let mintOutputs = 0;
+    let meltOutputs = 0;
+    for (let i = 1; i < tx.outputs.length; ++i) {
+      // All the outputs must also have standard scripts
+      const txOutput = tx.outputs[i];
+      if (!hasStandardScript(txOutput, true)) {
+        return false;
+      }
+
+      // All outputs must be either HTR or the created token
+      const validTokensList = [HATHOR_TOKEN_CONFIG.uid, tx.tx_id];
+      if (!validTokensList.includes(txOutput.token)) {
+        return false;
+      }
+
+      // Counting authority outputs
+      if (transaction.isTokenDataAuthority(txOutput.token_data)) {
+        switch (txOutput.value) {
+          case TOKEN_MINT_MASK:
+            ++mintOutputs;
+            break;
+          case TOKEN_MELT_MASK:
+            ++meltOutputs;
+            break;
+        }
+      }
+    }
+
+    // Validating maximum of 1 mint and/or melt outputs
+    if (mintOutputs > 1 || meltOutputs > 1) {
+      return false;
+    }
+
+    // Validating inputs
+    for (const input of tx.inputs) {
+      // All inputs must be HTR
+      if (input.token !== HATHOR_TOKEN_CONFIG.uid) {
+        return false;
+      }
+    }
+
+    // All conditions were met: this is a standard NFT creation transaction
+    return true;
+
+    /**
+     * Identifies if the informed output script size is standard
+     * @param {TxOutput} output The transaction output, already decoded
+     * @param {boolean} [onlyStandardScriptType=false] If true, the decoded type will also be validated
+     */
+    function hasStandardScript(output, onlyStandardScriptType = false) {
```

Comment:
I feel the confusion comes from the method in the python lib (https://github.com/HathorNetwork/python-hathorlib/blob/main/hathorlib/base_transaction.py#L623)

We started as a simple method but then added more validation. In order to keep compatibility (because it's a lib), we kept the same method name. But I agree with @r4mmer that it can be improved.

I feel we should create this method inside `Output` model.

Reply:
Thanks for clarifying these points about "standard" outputs and scripts. As Pedro mentioned, this function was basically translated here from the python lib.

The `isNFTToken` function was now fully moved into the `validateNftCreation`, under the `CreateTokenTransaction` class, and expands upon the existing `Transaction` and `CreateTokenTransaction` validations. Instead of mentioning _standard_ features, all of them are now described individually: script length, type and so on.

Reply:
Further improved this on [929f4e0](https://github.com/HathorNetwork/hathor-wallet-lib/pull/378/commits/929f4e02211186fbba08f68c86a2cfac6fa41770): now an Output can identify its script's type and validate its length accordingly.